ipaserver/secrets/kem.py

```diff
@@ -124,6 +126,12 @@ def _format_public_key(self, key):
             encoding=serialization.Encoding.DER,
             format=serialization.PublicFormat.SubjectPublicKeyInfo)
 
+    def _get_dn(self, usage, principal):
+        servicename, host = principal.split('@')[0].split('/')
```

Comment:
I'd have sworn there used to be a helper to split out service, host, realm but I couldn't find it quickly.

Reply:
There is some code in ``ipapython.kerberos`` but it doesn't implement what I need.

Reply:
Looks like it was removed in 4.4 so this is ok.